springboot/src/main/java/software/amazonaws/example/product/product/SpringBootSampleApplication.java

```diff
@@ -36,37 +32,12 @@
     "com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent$RequestIdentity",
     "com.amazonaws.xray.entities.TraceHeader$SampleDecision"
   })
+@TypeHint(
+  types = { Product.class, Products.class}, access = { TypeAccess.PUBLIC_CONSTRUCTORS, TypeAccess.PUBLIC_METHODS }
+)
 public class SpringBootSampleApplication {
 
   public static void main(String[] args) {
     SpringApplication.run(SpringBootSampleApplication.class, args);
   }
-
-  @Bean
```

Comment:
There's no need to declare beans with `@Bean` since classes are already annotated as `@Component`.

Alternatively, `@Component` annotation could be removed and `@Bean` definitions could stay.

Spring Cloud Function recommends using functional bean registration instead of `@Bean` or `@Component` for faster startup times. I did not go with it because I did not want to change too much of this code. This startup time improvement is probably not relevant for native image.